packages/world/src/modules/index/tables/ReverseMapping.sol

```diff
@@ -0,0 +1,111 @@
+    string[] memory _fieldNames = new string[](1);
+    _fieldNames[0] = "keysWithValue";
+    return ("ReverseMapping", _fieldNames);
+  }
+
+  /** Register the table's schema */
+  function registerSchema(uint256 _tableId) internal {
+    StoreSwitch.registerSchema(_tableId, getSchema(), getKeySchema());
+  }
+
+  /** Set the table's metadata */
+  function setMetadata(uint256 _tableId) internal {
+    (string memory _tableName, string[] memory _fieldNames) = getMetadata();
+    StoreSwitch.setMetadata(_tableId, _tableName, _fieldNames);
+  }
+
+  /** Register the table's schema for the specified store */
+  function registerSchema(uint256 _tableId, IStore _store) internal {
+    _store.registerSchema(_tableId, getSchema(), getKeySchema());
+  }
+
+  /** Set the table's metadata for the specified store */
+  function setMetadata(uint256 _tableId, IStore _store) internal {
+    (string memory _tableName, string[] memory _fieldNames) = getMetadata();
+    _store.setMetadata(_tableId, _tableName, _fieldNames);
+  }
+
+  /** Get keysWithValue */
+  function get(uint256 _tableId, bytes32 valueHash) internal view returns (bytes32[] memory keysWithValue) {
+    bytes32[] memory _primaryKeys = new bytes32[](1);
+    _primaryKeys[0] = bytes32((valueHash));
+
+    bytes memory _blob = StoreSwitch.getField(_tableId, _primaryKeys, 0);
+    return (SliceLib.getSubslice(_blob, 0, _blob.length).decodeArray_bytes32());
+  }
+
+  /** Get keysWithValue from the specified store */
+  function get(
+    uint256 _tableId,
+    IStore _store,
+    bytes32 valueHash
+  ) internal view returns (bytes32[] memory keysWithValue) {
+    bytes32[] memory _primaryKeys = new bytes32[](1);
+    _primaryKeys[0] = bytes32((valueHash));
+
+    bytes memory _blob = _store.getField(_tableId, _primaryKeys, 0);
+    return (SliceLib.getSubslice(_blob, 0, _blob.length).decodeArray_bytes32());
+  }
+
+  /** Set keysWithValue */
+  function set(uint256 _tableId, bytes32 valueHash, bytes32[] memory keysWithValue) internal {
+    bytes32[] memory _primaryKeys = new bytes32[](1);
+    _primaryKeys[0] = bytes32((valueHash));
+
+    StoreSwitch.setField(_tableId, _primaryKeys, 0, EncodeArray.encode((keysWithValue)));
+  }
+
+  /** Push an element to keysWithValue */
+  function push(uint256 _tableId, bytes32 valueHash, bytes32 _element) internal {
+    bytes32[] memory _primaryKeys = new bytes32[](1);
+    _primaryKeys[0] = bytes32((valueHash));
+
+    bytes memory _blob = StoreSwitch.getField(_tableId, _primaryKeys, 0);
+    bytes memory _newBlob = abi.encodePacked(_blob, abi.encodePacked((_element)));
+    StoreSwitch.setField(_tableId, _primaryKeys, 0, _newBlob);
+  }
+
+  /* Delete all data for given keys */
+  function deleteRecord(uint256 _tableId, bytes32 valueHash) internal {
+    bytes32[] memory _primaryKeys = new bytes32[](1);
+    _primaryKeys[0] = bytes32((valueHash));
+
+    StoreSwitch.deleteRecord(_tableId, _primaryKeys);
+  }
+}
```

Comment:
broader note/thought, esp because we ran into world contract size limits, is letting you conditionally specify in the mud config -> autogen which methods you need so we can have only what we need for the core libs

even better would be an optimization step before deploy that 1) finds all the calls to these libs in the contracts being deployed and 2) removes any unused calls/dead code before compiling

Reply:
Agreed! Sometimes we might still want the switch so the table can be used both from root systems/modules (= delegatecalled) and regular systems/modules (= called), but it's definitely useful to have a config option for those cases where it's clear how they'll be used.